### ISSUE TITLE
C# records wrong errors

### ISSUE THREAD

**Issue by VBAndCs** [dotnet/roslyn#49615]
Define this record:
` public record Student(string FirstName, string LastName);`

Now try:
```C#
            var x = new Student("","");
            x.FirstName = "";
            x.LastName = "";
```

and build the project. You will get the expected errors about init-only properties.

Now try to make any other error like changing the second line to:
`x.SetFirstName("");`
You will get the expected error:
> Error	CS1061	'Student' does not contain a definition for 'SetFirstName' and no accessible extension method 'SetFirstName' accepting a first argument of type 'Student' could be found.

Now, try to remove line 2 or change it back to:
`x.FirstName = "";`
Uou will still get the CS1061 error about the SetFirstName method! In fact this error will stuck with you as long as there is any error in the code! Even if you rebuild the solution, or change the record name to Student1! It only goes away if you fixed the errors, but comes back with any new error you type!

**Comment by Youssef1313** [comment]
This sounds like an IDE issue, not a compiler issue? Probably the Error List window is hanging or something like that.
cc: @mavasani

**Comment by mavasani** [comment]
Unable to repro this issue on the latest 16.9 Preview bits: 
![ErrorListBugReport](https://user-images.githubusercontent.com/10605811/103908002-d9022b00-50b6-11eb-88e8-c33646c95a0b.gif)
